### ISSUE TITLE
Azure Deployment Angular 7 ERROR @angular/compiler-cli needs to be 2.3.1 or greater. 

### ISSUE THREAD

**Issue by wenditurner** [angular/angular-cli#15898]
Error Azure Deployment Angular 7 @angular/compiler-cli needs to be 2.3.1 or greater. Current version is "7.2.15".

# 🐞 Bug report
### Command (mark with an `x`)
- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e

### Is this a regression?

This application was building and running nicely in Azure Cloud prior to v5.1 to v7.0 upgrade.  This has not yet successfully Azure Cloud Server.

### Description
This is an ASP.NET Core 2.1 Web MVC Spa upgrading from Angular 5.1 to Angular v7... with the hopes of stepping up to v8 but following the incremental approach of upgate.angular.io, I am at a locally running and functioning version of Angular v7.

This VS 2017 build and runs locally, all features work.  After checking-in to TFVS and deploying to dev environment in Azure Cloud for testing, I queue this build and receive this error which I do not receive locally.

<pre><code>
EXEC(0,0): Error : Version of @angular/compiler-cli needs to be 2.3.1 or greater. Current version is "7.2.15".
Cloud Development\RoadMap\Release\RoadMap\RoadMap.Web\RoadMap.Web.csproj(577,4): Error MSB3073: The command "node node_modules/webpack/bin/webpack.js --env.prod" exited with code 1.
Process 'msbuild.exe' exited with code '1'.
</code></pre>

## 🔬 Minimal Reproduction

`{
  "name": "app",
  "private": true,
  "version": "1.1.0",
  "scripts": {
    "test": "karma start App/test/karma.conf.js",
    "test_ci": "webpack --config webpack.config.vendor.js --env.prod && karma start App/test/karma.conf.nobrowser.js",
    "test_ondemand": "karma start App/test/karma.conf.nobrowser.js"
  },
  "dependencies": {
    "@angular/animations": "^7.0.0",
    "@angular/cli": "^8.3.4",
    "@angular/common": "^7.0.0",
    "@angular/compiler": "^7.0.0",
    "@angular/compiler-cli": "^7.0.0",
    "@angular/core": "^7.0.0",
    "@angular/forms": "^7.0.0",
    "@angular/upgrade": "~7.0",
    "@angular/platform-browser": "^7.0.0",
    "@angular/platform-browser-dynamic": "^7.0.0",
    "@angular/router": "^7.0.0",
    "@ngtools/webpack": "^1.9.1",
    "@progress/kendo-angular-buttons": "^5.0.0",
    "@progress/kendo-angular-common": "^1.0.0",
    "@progress/kendo-angular-dateinputs": "^4.0.0",
    "@progress/kendo-angular-dialog": "^4.0.0",
    "@progress/kendo-angular-dropdowns": "^4.0.0",
    "@progress/kendo-angular-excel-export": "^3.0.0",
    "@progress/kendo-angular-grid": "^4.0.0",
    "@progress/kendo-angular-inputs": "^6.0.0",
    "@progress/kendo-angular-intl": "^2.0.0",
    "@progress/kendo-angular-l10n": "^2.0.0",
    "@progress/kendo-angular-layout": "^4.0.0",
    "@progress/kendo-angular-pdf-export": "^2.0.0",
    "@progress/kendo-angular-popup": "^3.0.0",
    "@progress/kendo-angular-tooltip": "^2.0.0",
    "@progress/kendo-data-query": "^1.2.2",
    "@progress/kendo-drawing": "^1.5.12",
    "@progress/kendo-theme-default": "^2.47.0",
    "@types/webpack-env": "1.13.0",
    "angular2-template-loader": "0.6.2",
    "applicationinsights-js": "^1.0.14",
    "aspnet-prerendering": "^3.0.1",
    "aspnet-webpack": "^2.0.1",
    "awesome-typescript-loader": "3.2.1",
    "bootstrap": "^3.3.7",
    "css": "2.2.1",
    "css-loader": "0.28.4",
    "es6-shim": "0.35.3",
    "event-source-polyfill": "0.0.9",
    "expose-loader": "0.7.3",
    "extract-text-webpack-plugin": "2.1.2",
    "file-loader": "0.11.2",
    "html-loader": "0.4.5",
    "isomorphic-fetch": "2.2.1",
    "jquery": "^3.4.1",
    "json-loader": "0.5.4",
    "popper.js": "^1.12.9",
    "preboot": "4.5.2",
    "raw-loader": "0.5.1",
    "reflect-metadata": "0.1.10",
    "rxjs": "6.3.3",
    "rxjs-compat": "6.3.3",
    "style-loader": "0.18.2",
    "to-string-loader": "1.1.5",
    "tslint": "^5.0.0",
    "typescript": "^3.1.6",
    "url-loader": "0.5.9",
    "webpack": "2.5.1",
    "webpack-hot-middleware": "2.18.2",
    "webpack-merge": "4.1.0",
    "zone.js": "0.8.29"
  },
  "devDependencies": {
    "@types/applicationinsights-js": "^1.0.5",
    "@types/chai": "4.0.1",
    "@types/jasmine": "2.5.53",
    "babel-polyfill": "^6.26.0",
    "chai": "4.0.2",
    "core-js": "^2.5.3",
    "jasmine-core": "2.6.4",
    "karma": "1.7.0",
    "karma-chai": "0.1.0",
    "karma-chrome-launcher": "2.2.0",
    "karma-cli": "1.0.1",
    "karma-jasmine": "1.1.0",
    "karma-junit-reporter": "^1.2.0",
    "karma-phantomjs-launcher": "^1.0.4",
    "karma-webpack": "2.0.3",
    "node-sass": "^4.9.0",
    "sass-loader": "^6.0.6"
  }
}

## 🔥 Exception or Error
<pre><code>
EXEC(0,0): Error : Version of @angular/compiler-cli needs to be 2.3.1 or greater. Current version is "7.2.15".
Cloud Development\RoadMap\Release\RoadMap\RoadMap.Web\RoadMap.Web.csproj(577,4): Error MSB3073: The command "node node_modules/webpack/bin/webpack.js --env.prod" exited with code 1.
Process 'msbuild.exe' exited with code '1'.
</code></pre>

![10_21_2019_Angular_v7_Deployment_Build_Error](https://user-images.githubusercontent.com/4131666/67241257-f42c4080-f407-11e9-9e95-4037e3a6df6d.png)

## 🌍 Your Environment
<pre><code>
This is an ASP.NET Core 2.1 Web MVC Spa upgrading from Angular 5.1 to Angular v7... with the hopes of stepping up to v8 but following the incremental approach of upgate.angular.io, I am at a locally running and functioning version of Angular v7.
</code></pre>

**Anything else relevant?**

Current @angular/compiler-cli version 7.2.15 **_IS_** greater than Version 2.3.1 which confuses me.

See image:

**Comment by alan-agius4** [comment]
Hi, can you try to update `@ngtools/webpack` please?

**Comment by wenditurner** [comment]
I started this and I unraveled a rats nest of package version dependency issues, I'm doing this now... let you know.

**Comment by wenditurner** [comment]
Webpack v2.5 to v4.41 upgrade ERRORS (w/ Angular v5.1 to v7.0 upgrade)
(keep in mind, we are hoping to upgrade to Angular v8.2 ultimately but update.angular.io suggests an incremental upgrade 
and have not gotten to Angular v7 successfully - yet)

Having a bear of a time getting the upgrade of my Angular app from v5.1 to v8.2 to work due to move to webpack v4

process:
1. follow all the instructions on update.angular.io
2. delete folder node_modules
3. delete npm-shrinkwrap.json
4. npm install
5. npm shrinkwrap
-->> all good to here
6. node node_modules/webpack/bin/webpack --config webpack.config.vendor.js
-->>> **errors**

after step 4:

`
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN extract-text-webpack-plugin@3.0.2 requires a peer of webpack@^3.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN The package tslint is included as both a dev and production dependency.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

`
after step 6:

`
(node:20228) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\webpack\lib\Chunk.js:866
                throw new Error(
                ^

Error: Chunk.entrypoints: Use Chunks.groupsIterable and filter by instanceof Entrypoint instead
    at Chunk.get (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\webpack\lib\Chunk.js:866:9)
    at C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\extract-text-webpack-plugin\dist\index.js:176:48
    at Array.forEach (<anonymous>)
    at C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\extract-text-webpack-plugin\dist\index.js:171:18
    at AsyncSeriesHook.eval [as callAsync] (eval at create (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:7:1)
    at AsyncSeriesHook.lazyCompileHook (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\tapable\lib\Hook.js:154:20)
    at Compilation.seal (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\webpack\lib\Compilation.js:1342:27)
    at compilation.finish.err (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\webpack\lib\Compiler.js:675:18)
    at hooks.finishModules.callAsync.err (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\webpack\lib\Compilation.js:1261:4)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:24:1)
    at AsyncSeriesHook.lazyCompileHook (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\tapable\lib\Hook.js:154:20)
    at Compilation.finish (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\webpack\lib\Compilation.js:1253:28)
    at hooks.make.callAsync.err (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\webpack\lib\Compiler.js:672:17)
    at _err0 (eval at create (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:11:1)
    at _addModuleChain (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\webpack\lib\Compilation.js:1185:12)
    at processModuleDependencies.err (C:\dev\Cloud Development\Intake\Dev\1.1.0_Angular_6\Corvel.Intake.Web\node_modules\webpack\lib\Compilation.js:1097:9)
    at process._tickCallback (internal/process/next_tick.js:61:11)

`

I have tried to make changes to upgrade from "extract-text-webpack-plugin": "2.1.2" to "mini-css-extract-plugin": "^0.8.0",
but no matter what I've tried I still seem to require the extract-text-webpack-plugin due to another package, also mini-css-extract-plugin change in webpack are not entirely clear to me... is there an example other the generic given on their webpage?

(please see package.json v5.1 vs package.json v7.0 below)

I have gone and created a new little template app with:
ng new ngcli-webpack

My challenge is going the correct direction, Is there the equivalent update.angular.io for webpack update? 

I get that webpack v4 is now a part of Angular CLI v7 but ...
the list of exact instructions/changes from one version to the next is unclear.

Any help?  Suggestions?


******************
******************
v5.1 package.json
******************
******************

`
{
  "name": "Web",
  "private": true,
  "version": "1.0.0",
  "scripts": {
    "test": "karma start ClientApp/test/karma.conf.js",
    "test_ci": "webpack --config webpack.config.vendor.js --env.prod && karma start ClientApp/test/karma.conf.nobrowser.js",
    "test_ondemand": "karma start ClientApp/test/karma.conf.nobrowser.js"
  },
  "dependencies": {
    "@angular/animations": "^5.1.1",
    "@angular/common": "^5.1.1",
    "@angular/compiler": "^5.1.1",
    "@angular/compiler-cli": "^5.1.1",
    "@angular/core": "^5.1.1",
    "@angular/forms": "^5.1.1",
    "@angular/http": "^5.1.1",
    "@angular/platform-browser": "^5.1.1",
    "@angular/platform-browser-dynamic": "^5.1.1",
    "@angular/platform-server": "^5.1.1",
    "@angular/router": "^5.1.1",
    "@ngtools/webpack": "^1.9.1",
    "@progress/kendo-angular-buttons": "^4.0.0",
    "@progress/kendo-angular-dateinputs": "^2.2.0",
    "@progress/kendo-angular-dialog": "^3.4.0",
    "@progress/kendo-angular-dropdowns": "^2.1.0",
    "@progress/kendo-angular-excel-export": "^1.0.7",
    "@progress/kendo-angular-grid": "^3.0.0",
    "@progress/kendo-angular-inputs": "^2.2.0",
    "@progress/kendo-angular-intl": "^1.4.0",
    "@progress/kendo-angular-l10n": "^1.1.0",
    "@progress/kendo-angular-layout": "^3.1.0",
    "@progress/kendo-angular-popup": "^2.2.0",
    "@progress/kendo-angular-tooltip": "^0.4.0",
    "@progress/kendo-data-query": "^1.2.2",
    "@progress/kendo-drawing": "^1.5.2",
    "@progress/kendo-theme-default": "^2.47.0",
    "@types/webpack-env": "1.13.0",
    "angular2-template-loader": "0.6.2",
    "applicationinsights-js": "^1.0.14",
    "aspnet-prerendering": "^3.0.1",
    "aspnet-webpack": "^2.0.1",
    "awesome-typescript-loader": "3.2.1",
    "bootstrap": "^3.3.7",
    "css": "2.2.1",
    "css-loader": "0.28.4",
    "es6-shim": "0.35.3",
    "event-source-polyfill": "0.0.9",
    "expose-loader": "0.7.3",
    "extract-text-webpack-plugin": "2.1.2",
    "file-loader": "0.11.2",
    "html-loader": "0.4.5",
    "isomorphic-fetch": "2.2.1",
    "jquery": "3.2.1",
    "json-loader": "0.5.4",
    "popper.js": "^1.12.9",
    "preboot": "4.5.2",
    "raw-loader": "0.5.1",
    "reflect-metadata": "0.1.10",
    "rxjs": "^5.5.2",
    "style-loader": "0.18.2",
    "to-string-loader": "1.1.5",
    "typescript": "^2.5.3",
    "url-loader": "0.5.9",
    "webpack": "2.5.1",
    "webpack-hot-middleware": "2.18.2",
    "webpack-merge": "4.1.0",
    "zone.js": "0.8.12"
  },
  "devDependencies": {
    "@types/applicationinsights-js": "^1.0.5",
    "@types/chai": "4.0.1",
    "@types/jasmine": "2.5.53",
    "babel-polyfill": "^6.26.0",
    "chai": "4.0.2",
    "core-js": "^2.5.3",
    "jasmine-core": "2.6.4",
    "karma": "1.7.0",
    "karma-chai": "0.1.0",
    "karma-chrome-launcher": "2.2.0",
    "karma-cli": "1.0.1",
    "karma-jasmine": "1.1.0",
    "karma-junit-reporter": "^1.2.0",
    "karma-phantomjs-launcher": "^1.0.4",
    "karma-webpack": "2.0.3",
    "node-sass": "^4.9.0",
    "sass-loader": "^6.0.6"
  }
}

`

******************
******************
v7.0 package.json
******************
******************

`
{
  "name": "Web",
  "private": true,
  "version": "1.1.0",
  "scripts": {
    "test": "karma start ClientApp/test/karma.conf.js",
    "test_ci": "webpack --config webpack.config.vendor.js --env.prod && karma start ClientApp/test/karma.conf.nobrowser.js",
    "test_ondemand": "karma start ClientApp/test/karma.conf.nobrowser.js"
  },
  "dependencies": {
    "@angular/animations": "^7.0.0",
    "@angular/cli": "^7.0.1",
    "@angular/common": "^7.0.0",
    "@angular/compiler": "^7.0.0",
    "@angular/compiler-cli": "^7.0.0",
    "@angular/core": "^7.0.0",
    "@angular/forms": "^7.0.0",
    "@angular/http": "^7.0.0",
    "@angular/platform-browser": "^7.0.0",
    "@angular/platform-browser-dynamic": "^7.0.0",
    "@angular/router": "^7.0.0",
    "@ngtools/webpack": "^7.0.0",
    "@progress/kendo-angular-buttons": "^5.0.0",
    "@progress/kendo-angular-common": "^1.0.0",
    "@progress/kendo-angular-dateinputs": "^4.0.0",
    "@progress/kendo-angular-dialog": "^4.0.0",
    "@progress/kendo-angular-dropdowns": "^4.0.0",
    "@progress/kendo-angular-excel-export": "^3.0.0",
    "@progress/kendo-angular-grid": "^4.0.0",
    "@progress/kendo-angular-inputs": "^6.0.0",
    "@progress/kendo-angular-intl": "^2.0.0",
    "@progress/kendo-angular-l10n": "^2.0.0",
    "@progress/kendo-angular-layout": "^4.0.0",
    "@progress/kendo-angular-pdf-export": "^2.0.0",
    "@progress/kendo-angular-popup": "^3.0.0",
    "@progress/kendo-angular-tooltip": "^2.0.0",
    "@progress/kendo-data-query": "^1.2.2",
    "@progress/kendo-drawing": "^1.5.12",
    "@progress/kendo-theme-default": "^2.47.0",
    "@types/webpack-env": "1.13.0",
    "angular2-template-loader": "0.6.2",
    "applicationinsights-js": "^1.0.14",
    "aspnet-prerendering": "^3.0.1",
    "aspnet-webpack": "^3.0.0",
    "awesome-typescript-loader": "3.2.1",
    "bootstrap": "^3.3.7",
    "css": "2.2.1",
    "css-loader": "0.28.4",
    "es6-shim": "0.35.3",
    "event-source-polyfill": "0.0.9",
    "expose-loader": "0.7.3",
    "extract-text-webpack-plugin": "^3.0.0",
    "file-loader": "0.11.2",
    "html-loader": "0.4.5",
    "isomorphic-fetch": "2.2.1",
    "jquery": "^3.4.1",
    "json-loader": "0.5.4",
    "popper.js": "^1.12.9",
    "preboot": "4.5.2",
    "raw-loader": "0.5.1",
    "reflect-metadata": "0.1.10",
    "rxjs": "6.3.3",
    "rxjs-compat": "6.3.3",
    "style-loader": "0.18.2",
    "to-string-loader": "1.1.5",
    "tslint": "^5.0.0",
    "url-loader": "0.5.9",
    "webpack-dev-middleware": "^3.7.2",
    "webpack-hot-middleware": "2.18.2",
    "zone.js": "0.8.29"
  },
  "devDependencies": {
    "@types/applicationinsights-js": "^1.0.5",
    "@types/chai": "4.0.1",
    "@types/jasmine": "2.5.53",
    "babel-polyfill": "^6.26.0",
    "chai": "4.0.2",
    "core-js": "^2.5.3",
    "jasmine-core": "2.6.4",
    "karma": "1.7.0",
    "karma-chai": "0.1.0",
    "karma-chrome-launcher": "2.2.0",
    "karma-cli": "1.0.1",
    "karma-jasmine": "1.1.0",
    "karma-junit-reporter": "^1.2.0",
    "karma-phantomjs-launcher": "^1.0.4",
    "karma-webpack": "^4.0.0",
    "node-sass": "^4.9.0",
    "sass-loader": "^6.0.6",
    "tslint": "^5.0.0",
    "typescript": "^3.1.6",
    "webpack": "^4.41.2",
    "webpack-cli": "^3.3.9",
    "webpack-merge": "^4.1.0"
  }
}
`

**Comment by alan-agius4** [comment]
Duplicate of https://github.com/angular/angular-cli/issues/15926

**Comment by wenditurner** [comment]
You've completely closed this issue to any support or help.  ???
